railties/lib/rails/test_unit/line_filtering.rb

```diff
@@ -5,7 +5,7 @@
 module Rails
   module LineFiltering # :nodoc:
     def run(reporter, options = {})
-      options[:filter] = Rails::TestUnit::Runner.compose_filter(self, options[:filter])
+      options = options.merge(filter: Rails::TestUnit::Runner.compose_filter(self, options[:filter]))
```

Comment:
You could use `options.merge!` here @jonathanhefner

Reply:
That would defeat the purpose. :smile:  See https://github.com/rails/rails/pull/49141#discussion_r1317688054.

Reply:
🤦 yep it would.